src/core/data.js

```diff
@@ -0,0 +1,118 @@
+module.exports = class {
+  constructor(options, provider) {
+    this._processors = {};
+    this._provider = provider;
+    this._options = Object.assign(
+      {
+        startSymbol: "{{",
+        endSymbol: "}}"
+      },
+      options
+    );
+
+    if (this._options.startSymbol.length !== 2) {
+      throw new Error("The startSymbol should contains 2 charaters");
+    }
+
+    if (this._options.endSymbol.length !== 2) {
+      throw new Error("The endSymbol should contains 2 charaters");
+    }
+
+    const o = {
```

Comment:
🤓 I'm not against abbreviations in some cases (as iterator callbacks) but here it's really hard to guess what it?
Moreover, I think we could just use two variables (`start` and `end` ) as `o` is never used alone.